### PR TITLE
Programmer's documentation

### PR DESCRIPTION
**Pull request by vsenderov**
This is the current state of the programmer's documentation, found under `doc\Developer.md`. It is not quite complete, but I would like to merge it, as I will be focusing on the user documentation next.

### CODE REVIEW

File: treeppl/doc/Developer.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  - `flip.tppl`, which is the TreePPL program source,
+  - `data.mc`, which is a MCore source file, containig input as bindings (temporary measure),
+  - and `flip.mc`, which is a test CorePPL source file (same filename extension as MCore!); it contains the compiled-to-CorePPL instructions, to which the tests will compare the `flip.tppl` + `data.mc` after it has been compiled to CorePPL :)
+
+There are three ways to compile this program:
+
+  1. Build the `tppl` compiler, and use it to compile the program,
+  
+  ```
+  make build/tppl
```
Comment (larshum) [inline treeppl/doc/Developer.md:84]:
```suggestion
  make build/tpplc
```

File: treeppl/doc/Developer.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  build/tppl models/flip/flip.tppl models/flip/data.mc
```
Comment (larshum) [inline treeppl/doc/Developer.md:85]:
```suggestion
  build/tpplc models/flip/flip.tppl models/flip/data.mc
```

File: treeppl/doc/Developer.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+
+## Components of `tppl`
+
+### Compiler command-line `src/tppl.mc`
+
+The current command-line arguments of `tppl` are the input TreePPL program and the input MCore data.  E.g.:
+
+```
+tppl PROGRAM.tppl DATA.mc
```
Comment (larshum) [inline treeppl/doc/Developer.md:147]:
```suggestion
tpplc PROGRAM.tppl DATA.mc 
```

File: treeppl/doc/Developer.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+In certain circumstances you may need to install the CUDA C++ backend compiler itself, RootPPL `rppl`.
+
+```
+make install-rootppl
+```
+
+### How does compilation work?
+
+To get started with hacking on `tppl` you need to first understand how compilation works.
```
[the remark this answers is not in the document]
Comment (david-broman) [reply treeppl/doc/Developer.md:71]: "Hi and welcome" sounds a bit too informal.

Also, I do not think we should have author information in text files in general. We do not have any author information for other parts in the Miking project. It gets complicated when several people edit. We can see this in the history anyway in Github. We should add a list with contributors in general somewhere.

**Review by david-broman — COMMENTED** [review]
There are many comments and TODOs in the text. I'm not sure we want to have that in the text if we commit it.

**Comment by dlunde** [comment]
@vsenderov I guess this PR is now deprecated? We should perhaps move it to the TreePPL repository?

**Comment by vsenderov** [comment]
yes, this is deprecated. this stuff is not in the "webpage" on a separate
branch

**Comment by dlunde** [comment]
So we should simply close this PR?

**Comment by vsenderov** [comment]
Yes!

**Pull request closed without merging**